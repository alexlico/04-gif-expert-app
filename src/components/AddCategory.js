import React, { useState } from 'react'
import PropTypes from 'prop-types'
export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('')
    
    const handlerInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handlerSubmit = (e) => {
        e.preventDefault();
        if(inputValue.trim().length > 2){
            setCategories((cat) => [inputValue, ...cat])
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handlerSubmit}>
            <input 
                type="text" 
                value={inputValue}
                onChange={handlerInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}