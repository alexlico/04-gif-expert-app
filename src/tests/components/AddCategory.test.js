import {shallow} from 'enzyme'
import { AddCategory } from '../../components/AddCategory';
describe('Pruebas sobre el componente <AddCategory/>', () => {
    
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />)

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />)
    })

    test('Debe mostrar el componente correctamente', () => {    
        expect(wrapper).toMatchSnapshot()
    })
    
    test('Debe cambiar el contenido de la caja de texto', () => {    
        const value = 'Goku';
        wrapper.find('input').simulate('change', { target:{value} } )
        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find('input').prop('value').trim()).toBe(value)    
    })

    test('No debe enviar la informacion del submit', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    })
    
    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = 'Goku';

        wrapper.find('input').simulate('change', { target: {value} });
        wrapper.find('form').simulate('submit', { preventDefault(){} })

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value').trim()).toBe('');

    })
})
