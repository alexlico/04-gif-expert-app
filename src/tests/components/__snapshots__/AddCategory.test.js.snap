// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente <AddCategory/> Debe cambiar el contenido de la caja de texto 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
    Goku
  </p>
  <input
    onChange={[Function]}
    type="text"
    value="Goku"
  />
</form>
`;

exports[`Pruebas sobre el componente <AddCategory/> Debe mostrar el componente correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
